pyiceberg/expressions/__init__.py

```diff
@@ -701,7 +701,7 @@ def bind(self, schema: Schema, case_sensitive: bool = True) -> BoundLiteralPredi
 
     def __eq__(self, other: Any) -> bool:
         """Return the equality of two instances of the LiteralPredicate class."""
-        if isinstance(other, LiteralPredicate):
+        if isinstance(other, self.__class__):
```

Comment:
This fix is for the `LiteralPredicate`, but there are more:

- Line 367 should also use `self.__class__` for `IsNull`, `IsNaN` etc.
- Can you add `self.__class__` to the `SetPredicate` on line 534.
- The `__eq__` of `NotIn` on line 667 can go (`In` doesn't have one either).

Reply:
Raised a PR https://github.com/danielcweeks/iceberg-python/pull/1 to get these fixed :)